packages/react/src/components/List/ListContent/ListContent.jsx

```diff
@@ -167,7 +167,15 @@ const ListContent = ({
                 name={item.value}
                 data-testid={`${item.id}-checkbox`}
                 labelText=""
-                onClick={() => handleSelect(item.id, parentId)}
+                onClick={(evt) => {
+                  // after the carbon upgrade to 10.47 this event
+                  // is being fired twice, once on the label and once on the
+                  // input, so filtering out the LABEL event here and only firing on
+                  // the input.
+                  if (evt.target.tagName === 'INPUT' && !isLocked) {
```

Comment:
The argument could be made this should fire on label and not input. I chose input in this case, because many of our tests are using testIds tied to the checkbox element. Setting this to fire on the label breaks those tests. This is a carbon bug. I've filed an issue. It's re-creatable in their own storybook for the checkbox playground. https://github.com/carbon-design-system/carbon/issues/10122

Reply:
switching this to an onChange seems to be fine and tests all still pass. We need to double check the stories though.